src/installer/image/installer.go

```diff
@@ -44,15 +46,15 @@ func GetDigest(uri string) (string, error) {
 	return canonRef.Digest().Encoded(), nil
 }
 
-func NewImageInstaller(fs afero.Fs, props *Properties) installer.Installer {
+func NewImageInstaller(fs afero.Fs, props *Properties, pullSecret corev1.Secret) (installer.Installer, error) {
```

Comment:
bit misleading, someone might think that the `pullSecret` is actually a complete secret, but its created with `dynakube.PullSecretWithoutData`

you already have `Dynakube` in the properties, so you don't need to pass this